cpp/src/arrow/flight/sql/odbc/tests/connection_test.cc

```diff
@@ -246,10 +227,9 @@ TYPED_TEST(ODBCTestBase, TestSQLDriverConnect) {
 
   // Connecting to ODBC server.
   ASSERT_EQ(SQL_SUCCESS,
-            SQLDriverConnect(conn, NULL, &connect_str0[0],
+            SQLDriverConnect(this->conn, NULL, &connect_str0[0],
                              static_cast<SQLSMALLINT>(connect_str0.size()), out_str,
-                             kOdbcBufferSize, &out_str_len, SQL_DRIVER_NOPROMPT))
-      << GetOdbcErrorMessage(SQL_HANDLE_DBC, conn);
```

Comment:
How come we are removing `<< GetOdbcErrorMessage(SQL_HANDLE_DBC, conn);`? This is what prints out the error message when things fail.

Reply:
Added the checks back. I don't have a strong opinion on this one. Originally I didn't want to flood the terminal with the error messages since they could be the same errors, but `GetOdbcErrorMessage` will still probably be helpful for debugging, especially when a single test is run.